src/app/lib/select/select.component.ts

```diff
@@ -1,22 +1,56 @@
-import { Component, Input, OnInit } from '@angular/core';
+import { Component, Input, OnInit, forwardRef } from '@angular/core';
+import { ControlValueAccessor, NG_VALUE_ACCESSOR } from '@angular/forms';
 
 @Component({
   selector: 'hc-select',
   templateUrl: './select.component.html',
-  styleUrls: ['./select.component.scss']
+  styleUrls: ['./select.component.scss'],
+  providers: [
+    {
+      provide: NG_VALUE_ACCESSOR,
+      useExisting: forwardRef(() => SelectComponent),
+      multi: true
+    }
+  ]
 })
-export class SelectComponent implements OnInit {
+export class SelectComponent implements ControlValueAccessor {
 
   @Input() placeholder: string;
-  @Input() options = [];
+  @Input() options: Array<string> = [];
   @Input() disabled: boolean = false;
-  alpha = 1.0;
+  disabledAlpha = 0.4;
+  enabledAlpha = 1.0;
+  _value: string;
+  onChange: any = () => { };
+  onTouched: any = () => { };
+
+  get alpha() {
+    return this.disabled ? this.disabledAlpha : this.enabledAlpha;
+  }
 
-  constructor() { }
+  get value() {
+    return this._value;
+  }
+
+  set value(val: string) {
+    this._value = val;
```

Comment:
Do we need to check for the existence of onChange and onTouch here or is it guaranteed that they will be set by this point?

Reply:
This would get called when the user has selected a new element and the change event is dispatched: `(change)="value = $event.target.value"`. onChange and onTouch are initialized with the component and so I believe it's guaranteed they'll be set by the time the user chooses a new selection.